### PR TITLE
Finalizing the Project

### PR DESCRIPTION
**Pull request by Khaled-AlKhateeb**
#### Logic
- Each class from the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png) should be created in a separate .rb file.
- All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes should be implemented.
- All properties visible in the diagram should be defined and set up in the constructor method. _Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created._
- All methods visible in the diagram should be implemented:
  - add_item method (in all classes that have that method)
    - should take an instance of the Item class as an input.
    - should add the input item to the collection of items.
    - should add self as a property of the item object (by using the correct setter from the item object).
  - can_be_archived?() in the Item class
    - should return true if published_date is older than 10 years.
    - otherwise, it should return false.
  - move_to_archive() in the Item class
    - should reuse can_be_archived?() method.
    - should change the archived property to true if the result of the can_be_archived?() method is true.
    - should do nothing if the result of the can_be_archived?() method is false.
  - can_be_archived?() in the Book class
    - should override the method from the parent class.
    - should return true if parent's method returns true OR if cover_state equals to "bad".
    - otherwise, it should return false.
  - can_be_archived?() in the MusicAlbum class
    - should override the method from the parent class.
    - should return true if parent's method returns true AND if on_spotify equals true.
    - otherwise, it should return false.
  - can_be_archived?() in the Movie class
    - should override the method from the parent class.
    - should return true if parent's method returns true OR if silent equals true.
    - otherwise, it should return false.
  - can_be_archived?() in the Game class
    - should override the method from the parent class.
    - should return true if parent's method returns true AND if last_played_at is older than 2 years.
    - otherwise, it should return false.
  - Add unit tests for all implemented methods.
#### User interface
- Create a main.rb file that will serve as your console app entry-point.
- Your console app, at the start, should:
  - Present the user with a list of options to perform.
  - Let users choose an option.
  - If needed, ask for parameters for the option.
  - Have a way to quit the app.
- The following options should be available:
  - List all books
  - List all music albums
  - List all movies
  - List of games
  - List all genres (e.g 'Comedy', 'Thriller')
  - List all labels (e.g. 'Gift', 'New')
  - List all authors (e.g. 'Stephen King')
  - List all sources (e.g. 'From a friend', 'Online shop')
  - Add a book
  - Add a music album
  - Add a movie
  - Add a game
- All data should be preserved by saving collections in .json files.
#### Database
- Create a schema.sql file with tables that will be analogical to the structure of the classes in your app. As you cannot implement inheritance in the database tables - add all properties and associations from the parent Item class as table columns to all tables based on the child classes. Thanks to that it can be used to store data in the future.

### CODE REVIEW

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @Khaled-AlKhateeb, 

Good job so far! 🔥
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes ♻
- [x] You should have recorded a video presenting the features of the project with these requirements:
1. The maximum length of the video is 5 minutes.
2. The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
3. All the members in the video should appear in the video and talk about the project.
4. The video contains a highlight about a piece of the code the student finds interesting or is proud of making.

Let's add it

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md [hunk 1]
```diff
@@ -87,6 +87,7 @@ The app help you to keep recods about any items you own, thing like
 ## Video Description <a name="vd-disc"></a>
 
 In this recorder video we are describing how we built this amazing project as a team.
+[Video](https://drive.google.com/file/d/1sdiqXFGtH41n97M2af-3nTxuj00K2qES/view?usp=sharing)
```
Comment (codecaiine) [inline README.md:90]: - [ ] Good job presentation :clap: But I noticed that the length of your presentation video is too long (9 minutes)  according to the project requirement, **the maximum length of the video is 5 minutes**. Please make sure that your presentation video length is a maximum of 5  minutes as required for this project, you don't need to cover everything but the video contains a highlight about a piece of the code you find interesting or proud of making. :+1:

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @Khaled-AlKhateeb @johnkioko @OpondoG , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Great project implementation :heavy_check_mark: 
- No rubocop errors :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — APPROVED** [review]
### Hi  @team🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**